cvmfs/publish/repository_abort.cc

```diff
@@ -26,11 +26,7 @@ void Publisher::WipeScratchArea() {
 }
 
 void Publisher::Abort() {
-  if (is_publishing_.IsLocked()) {
-    throw EPublish(
-      "Repository " + settings_.fqrn() + " is currently publishing "
-      "(aborting abort)", EPublish::kFailTransactionState);
-  }
+  ServerLockFileGuard g(is_publishing_);
```

Comment:
I'm not sure this is correct: the new code would acquire the lock instead of checking if another process holds it, isn't it?

If another process holds it, it will throw an exception as the old code. But unlike the old code, the `abort` command now briefly holds the `is_publishing` lock, which is at least surprising.

Reply:
Yes, this is correct and deliberate.  The old code has a race condition: it will check the lock at a point in time, but there is nothing that prevents another publisher action from starting while the abort is in progress.  The new code ensures that nothing else tries to modify the repository state mid-abort.

Reply:
Ok, thanks for the clarification!